python/TestHarness/util.py

```diff
@@ -146,86 +154,94 @@ def formatResult(tester_data, result, options, color=True):
     color_opts = {'code' : options.code, 'colored' : options.colored}
 
     # container for every printable item
-    formatted_result = { 'p' : None,
-                         'n' : None,
-                         'c' : None,
-                         'j' : None,
-                         's' : None,
-                         't' : None}
-
-    # Decorate pre-result
-    if 'p' in terminal_format:
-        # 8 characters is the current max length of any of our statuses
-        formatted_result['p'] = (' '*(8-len(status.status)) + status.status, status.color)
-
-    # Support the option of no result...
-    if 's' in terminal_format:
-        # If result is empty, default to using the testers status message as the result
-        if not result:
-            formatted_result['s'] = (str(tester.getStatusMessage()), status.color)
-        else:
-            formatted_result['s'] = (result, status.color)
-
-    # Support the option of no name...
-    if 'n' in terminal_format:
-        formatted_result['n'] = (tester.getTestName(), None)
-
-    # Decorate and format timing
-    if options.timing and 't' in terminal_format:
-        actual = float(tester_data.getTiming())
-        int_len = len(str(int(actual)))
-        precision = min(3, max(0,(4-int_len)))
-        formatted_result['t'] = ('[' + '{0: <6}'.format('%0.*fs' % (precision, actual)) + ']', None)
-
-    # Special Case: When printing a pre-status _and_ a status, strip out the status
-    # if that status is identical to the pre-status. This is so we can support the
-    # printing of TestHarness statuses like RUNNING... while not printing two 'OK's
-    if (formatted_result['p'] and formatted_result['s']) \
-       and (formatted_result['p'][0].replace(' ', '') == formatted_result['s'][0].replace(' ', '')):
-        formatted_result['s'] = None
-
-    # Format the caveats if any
-    if tester.getCaveats() and 'c' in terminal_format:
+    formatted_results = {}
+    for format_key in terminal_format:
+        formatted_results[format_key] = None
```

Comment:
These 3 lines can be done with:
`formatted_results = dict.fromkeys(terminal_format)`